dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04305UpdateWorkflowActionTable.java

```diff
@@ -280,27 +286,44 @@ private void createWorkflowActionStepTable(final DotConnect dc) throws DotDataEx
 
             Logger.info(this, "Table 'workflow_action_step' does not exists, creating it");
             needToCreate = true;
+            // in some databases if an error is throw the transaction is not longer valid
+            this.closeAndStartTransaction();
         }
 
         if (needToCreate) {
             try {
-                dc.executeStatement(createIntermediateTable());
+                dc.setSQL(createIntermediateTable()).loadResult();
                 // The SQL Server and Oracle table definition already include de PK creation
                 if (DbConnectionFactory.isMySql() || DbConnectionFactory.isPostgres()) {
-                    dc.executeStatement(createIntermediateTablePk());
+                    dc.setSQL(createIntermediateTablePk()).loadResult();
                 }
 
                 // adding the FK
                 Logger.info(this, "Creating the Workflow action step intermediate FKs.");
-                dc.executeStatement(this.createIntermediateTableForeignKeyActionId());
-                dc.executeStatement(this.createIntermediateTableForeignKeyStepId());
-            } catch (SQLException e) {
+                dc.setSQL(this.createIntermediateTableForeignKeyActionId()).loadResult();
+                dc.setSQL(this.createIntermediateTableForeignKeyStepId()).loadResult();
+            } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)